src/main/java/gregtech/integration/forestry/recipes/CombRecipes.java

```diff
@@ -0,0 +1,493 @@
+                .inputs(combStack)
+                .duration(duration)
+                .EUt(volt.getCentrifugeEnergy());
+
+        int numGTOutputs = 0;
+        for (int i = 0; i < item.length; i++) {
+            if (item[i] == null || item[i] == ItemStack.EMPTY) continue;
+            // Add to Forestry Map
+            product.put(item[i], chance[i] / 10000.0f);
+            // Add to GregTech Map
+            if (numGTOutputs < RecipeMaps.CENTRIFUGE_RECIPES.getMaxOutputs()) {
+                if (chance[i] >= 10000) {
+                    builder.outputs(item[i]);
+                } else {
+                    builder.chancedOutput(item[i], chance[i], 0);
+                }
+                numGTOutputs++;
+            }
+        }
+
+        // Finalize Forestry Map
+        if (volt.compareTo(Voltage.MV) < 0 || !ForestryConfig.harderGTCombRecipes) {
+            if (ModuleFactory.machineEnabled(MachineUIDs.CENTRIFUGE)) {
+                RecipeManagers.centrifugeManager.addRecipe(duration, combStack, product.build());
+            }
+        }
+        // Finalize GregTech Map
+        builder.buildAndRegister();
+    }
+
+    private enum Voltage {
+        ULV, LV, MV, HV, EV, IV, LUV, ZPM, UV, UHV, UEV, UIV, UXV, OPV, MAX;
+
+        public int getVoltage() {
+            return (int) GTValues.V[ordinal()];
+        }
+
+        public int getChemicalEnergy() {
+            return getVoltage() * 3 / 4;
+        }
+
+        public int getAutoclaveEnergy() {
+            return (int) ((getVoltage() * 3 / 4) * Math.max(1, Math.pow(2, 5 - ordinal())));
+        }
+
+        public int getCentrifugeEnergy() {
+            return this == Voltage.ULV ? 5 : (getVoltage() / 16) * 15;
+        }
+
+        public int getChemicalTime() {
+            return 64 + ordinal() * 32;
+        }
+
+        public int getCentrifugeTime() {
+            if (ForestryConfig.harderGTCombRecipes) {
+                return 128 * (Math.max(1, ordinal()));
+            } else {
+                return 96 + ordinal() * 32;
+            }
+        }
+
+        public FluidStack getFluid() {
+            if (this.compareTo(Voltage.MV) < 0) {
+                return Materials.Water.getFluid((this.compareTo(Voltage.ULV) > 0) ? 1000 : 500);
+            } else if (this.compareTo(Voltage.HV) < 0) {
+                return Materials.DistilledWater.getFluid(1000);
+            } else if (this.compareTo(Voltage.EV) < 0) {
+                return Materials.SulfuricAcid.getFluid(GTValues.L);
+            } else if (this.compareTo(Voltage.IV) < 0) {
+                return Materials.HydrochloricAcid.getFluid(GTValues.L);
+            } else if (this.compareTo(Voltage.LUV) < 0) {
+                return Materials.HydrofluoricAcid.getFluid((int) (Math.pow(2, this.compareTo(Voltage.HV)) * GTValues.L));
+            } else {
+                return Materials.FluoroantimonicAcid.getFluid((int) (Math.pow(2, this.compareTo(Voltage.LUV)) * GTValues.L));
```

Comment:
I think these should use `125` or `150` instead of `GTValues.L`. It feels off that they use ingot-based quantities for non-ingot fluids.